core/rint/test/TRintTests.cxx

```diff
@@ -0,0 +1,41 @@
+#include <iostream>
+#include <string>
+
+#include "gtest/gtest.h"
+
+#include "TRint.h"
+
+TEST(TRint, UnrecognizedOptions)
+{
+   // Create array of options
+   int argc{4};
+   char **argv = new char *[argc];
+   argv[0] = const_cast<char *>("-q");
+   argv[1] = const_cast<char *>("-z");
+   argv[2] = const_cast<char *>("--nonexistingoption");
+   argv[3] = const_cast<char *>("-b");
+
+   // Test the error reported on std::cerr
+   std::cerr << std::flush;
+   // Redirect stderr
+   std::streambuf *old_cerr_streambuf = std::cerr.rdbuf();
+   std::ostringstream stderrstream;
+   std::cerr.rdbuf(stderrstream.rdbuf());
+   // Initializing TRint will delete from argv the recognized options leaving only unrecognized ones.
+   TRint app("App", &argc, argv);
+   // Restore old stderr
+   std::cerr.rdbuf(old_cerr_streambuf);
```

Comment:
you can use `testing::internal::CaptureStderr` to make this more readable, see e.g. `dataframe_cache.cxx` (we rely on those internal interfaces of googletest in a bunch of places, most importantly in `ROOTUnitTestSupport.h`, so you would not be introducing more coupling than we already have)

Reply:
Ok thanks a lot for the hint! Will rewrite this part of the test